### DISCUSSION TITLE
Query Regarding Vulnerability in @cypress/request - I want to maintain current version - 9.5.4

### DISCUSSION THREAD

**Discussion by Nguyenanh2011** [cypress-io/cypress#27661] [Questions and Help]
Dear Cypress Team,

I hope this email finds you well. I am reaching out to your esteemed team regarding a recent issue I've encountered while using the Cypress testing framework. I have thoroughly reviewed the documentation and explored various solutions, but I am seeking your expert advice to ensure the best course of action.

The issue revolves around a vulnerability in the @cypress/request package, specifically associated with Server-Side Request Forgery (SSRF) as indicated in advisory GHSA-p8p7-x288-28g6. The advisory recommends running npm audit fix --force to address the issue; however, this approach would lead to installing cypress@4.2.0, which introduces breaking changes. 
Additionally, this update affects compatibility with other packages, such as @cypress/request-promise, as it requires a newer version of @cypress/request.

Given these considerations, we are seeking guidance on how best to proceed. We have a few questions:

Are there any alternative solutions available that address the vulnerability in @cypress/request without requiring a breaking update to cypress@4.2.0?
Could you provide insights into the severity of the vulnerability and whether there are immediate risks that we should address?
 Our team's priority is to maintain our current Cypress version, which is 9.5.4, due to compatibility considerations, while minimizing the potential disruption to our existing test suite. Your expertise and guidance in this matter would be invaluable to our team.

Thank you in advance for your assistance. We greatly appreciate your dedication to supporting the Cypress community and look forward to your insights.

**Comment by portyu9** [comment]
Hi @Nguyenanh2011 

I would **not** use `npm audit fix --force` if it wants to replace Cypress 9.5.4 with Cypress 4.2.0. That is npm choosing a dependency graph that removes the advisory, not a sensible Cypress upgrade path.

The advisory affects:

- `@cypress/request <= 2.88.12`
- patched in `@cypress/request 3.0.0`

It is rated **Moderate (CVSS 6.1)** and involves bypassing SSRF protections through a cross-protocol redirect controlled by an attacker.

The supported solution is to upgrade Cypress rather than trying to preserve Cypress 9.5.4 indefinitely. Cypress 9 is now very old, so I would upgrade major-by-major and run the suite at each step.

You *could* experiment with an npm `overrides` entry forcing a newer `@cypress/request`, but that crosses a major-version boundary of a Cypress transitive dependency. Cypress 9.5.4 was not tested with that dependency combination, so I would treat that only as a temporary, thoroughly regression-tested workaround—not as the supported remediation.

The practical risk depends on whether untrusted URLs or attacker-controlled redirects can reach the vulnerable request path in your environment, but I would not dismiss the advisory solely because Cypress is "only used for testing."

Please mark this answer as accepted if it helped, thank you.
